File: C#/05 - Cars Assemble/CarsAssemble.cs
static class AssemblyLine
{
    public static double SuccessRate(int speed)
    {
        if (speed == 0) return 0.0;
        if (speed <= 4) return 1.0;
        if (speed <= 8) return 0.9;
        if (speed == 9) return 0.8;
        if (speed == 10) return 0.77;
        return 0.0;
    }
    
    public static double ProductionRatePerHour(int speed)
    {
        double successRate = SuccessRate(speed);
        return speed * 221.0 * successRate;
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        double productionRatePerHour = ProductionRatePerHour(speed);
        return (int)(productionRatePerHour / 60.0);
    }
}

File: Java/04 - Cars Assemble/CarsAssemble.java
public class CarsAssemble {
    
    private static final int CARS_PRODUCED_PER_HOUR = 221;


    public double productionRatePerHour(int speed) {
        double successRate;
        if (speed >= 1 && speed <= 4) {
            successRate = 1.0;
        } else if (speed <= 8) {
            successRate = 0.9;
        } else if (speed == 9) {
            successRate = 0.8;
        } else {
            successRate = 0.77;
        }
        return speed * CARS_PRODUCED_PER_HOUR * successRate;
    }

    public int workingItemsPerMinute(int speed) {
        return (int) (productionRatePerHour(speed)/60);
    }
}
